src/views/tls/tls-version/index.js

```diff
@@ -4,45 +4,88 @@ import {CardContent} from '@material-ui/core';
 import CardHeader from '@material-ui/core/CardHeader';
 import ChartContainer from 'common/chart-container';
 import CircularProgress from '@material-ui/core/CircularProgress';
+import MenuItem from '@material-ui/core/MenuItem';
 import MuiAlert from '@material-ui/lab/Alert';
 import Paper from '@material-ui/core/Paper';
 import {ResponsivePie} from '@nivo/pie';
 import Snackbar from '@material-ui/core/Snackbar';
 import Tab from '@material-ui/core/Tab';
 import Tabs from '@material-ui/core/Tabs';
 import React, {useEffect, useState} from 'react';
+// eslint-disable-next-line no-use-before-define
+import Select from '@material-ui/core/Select';
 
 /**
  * This component shows a pie chart with the number of requests
  * per TLS version in the sample data requests table.
  *
  * @return {ReactNode} TLS versions component
  */
+
 const TLSversion = () => {
   const [loading, setLoading] = useState(true);
   const [snackOpen, setSnackOpen] = useState(false);
   const [data, setData] = useState([]);
-  const [table, setTable] = useState('httparchive.smaller_sample_requests');
-
+  const [year, setYear] = useState('sample');
+  const [month, setMonth] = useState('01');
+  const [months, setMonths] = useState([
+    '01',
+    '02',
+    '03',
+    '04',
+    '05',
+    '06',
+    '07',
+    '08',
+    '09',
+    '10',
+    '11',
+    '12',
+  ]);
   useEffect(() => {
     setLoading(true);
     api
-      .get('/api/tls/tls-version?table=' + table)
+      .get('/api/tls/tls-version?year=' + year + '&month=' + month)
       .then((response) => {
         setLoading(false);
         setData(response.data.result);
       })
       .catch((err) => {
         setSnackOpen(true);
       });
-  }, [table]);
+    if (year === 'sample') {
+      setMonths(['Sample']);
+    } else if (year === '2018') {
+      setMonths(['04', '05', '06', '07', '08', '09', '10', '11', '12']);
```

Comment:
Does sending the list of every month cause an error in the backend? Just to be sure, usually, there is no error, and you handle the empties in a different way. I feel `months` more like a constant.

Reply:
It shows an empty graph when a previous month is selected, but the query runs anyway

Reply:
Since it is not throwing an error, it might be great if we add an empty state for these cases.

Go to `go/moonbase` to search for illustrations for those empty states, and change the state logic.

Thea idea is to let users input any date, but if it's empty, doesn't really matter as long as we show the correct message.

Reply:
Thanks!

Reply:
Done